napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -23,7 +24,16 @@
 from ..widgets.qt_mode_buttons import QtModePushButton, QtModeRadioButton
 from .qt_layer_controls_base import QtLayerControls
 
-INT32_MAX = 2 ** 31 - 1
+
+def double_as_uint_spinbox(dtype: np.dtype):
+    iinfo = np.iinfo(dtype)
+
+    spinbox = QDoubleSpinBox()
+    spinbox.setSingleStep(1)
+    spinbox.setMinimum(max(iinfo.min, 0))
```

Comment:
do we really want to always clip the min at 0?  certainly makes sense for labels, but could we make this a bit more reusable by not doing that?  or do we lose something?